zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NotebookAuthorization.java

```diff
@@ -65,7 +65,7 @@
 
   private NotebookAuthorization() {}
 
-  public static NotebookAuthorization init(ZeppelinConfiguration config) {
+  public static synchronized NotebookAuthorization init(ZeppelinConfiguration config) {
```

Comment:
Thanks @felixcheung & @AhyoungRyu I have reverted and will do it in another ticket.